spec/templater_spec.rb

```diff
@@ -164,11 +164,13 @@ def expect_environment_dependent_parameters_present(magic_namespace)
       expect(genders_config.non_existent).to eq([])
     end
 
-    before do
-      # Stub this so mock hunter config used.
-      stub_const('Metalware::Constants::HUNTER_PATH', TEST_HUNTER_PATH)
+    let :filesystem {
+      FileSystem.setup do |fs|
+        fs.with_repo_fixtures 'repo'
+      end
+    }
 
-      SpecUtils.use_unit_test_config(self)
+    before do
       SpecUtils.use_mock_determine_hostip_script(self)
       SpecUtils.use_mock_genders(self)
```

Comment:
I agree it would be nice to handle these in tests a better way, but I don't think `FakeFS` helps us here.

For the `determine-hostip` script, it doesn't look like `FakeFS` changes Ruby functions for running external commands to use the `FakeFS` - this is probably good in general, I imagine there would be a lot of odd cases and surprising behaviour if it did.

And then for using `genders`, we never actually load the `genders` file directly ourselves, rather we rely on `nodeattr` to parse it and query that for the results. Even if we loaded the `genders` file into the `FakeFS`, there would be no way to tell `nodeattr` to use this - the `FakeFS` isn't even a real file system; you'd need to have `nodeattr`, a non-Ruby program, somehow load the `genders` file from Ruby's memory!

There may be some improved abstraction we can use to paper over this sort of thing, but I agree we can do that in future as needed.